DQM/SiTrackerPhase2/plugins/Phase2TrackerMonitorDigi.cc

```diff
@@ -699,25 +691,76 @@ void Phase2TrackerMonitorDigi::bookLayerHistos(DQMStore::IBooker& ibooker,
                                                  Parameters.getParameter<double>("xmax"));
       else
         local_mes.ChargeOfDigis = nullptr;
-
-      edm::ParameterSet WidthParameters = config_.getParameter<edm::ParameterSet>("ClusterWidthH");
-      HistoName.str("");
-      HistoName << "ChargeOfDigisVsWidth_" << fname2.str();
-      if (Parameters.getParameter<bool>("switch") && WidthParameters.getParameter<bool>("switch"))
-        local_mes.ChargeOfDigisVsWidth = ibooker.book2D(HistoName.str(),
-                                                        HistoName.str(),
-                                                        Parameters.getParameter<int32_t>("Nbins"),
-                                                        Parameters.getParameter<double>("xmin"),
-                                                        Parameters.getParameter<double>("xmax"),
-                                                        WidthParameters.getParameter<int32_t>("Nbins"),
-                                                        WidthParameters.getParameter<double>("xmin"),
-                                                        WidthParameters.getParameter<double>("xmax"));
-      else
-        local_mes.ChargeOfDigisVsWidth = nullptr;
+      
+      if (clsFlag_){
+	edm::ParameterSet WidthParameters = config_.getParameter<edm::ParameterSet>("ClusterWidthH");
+	HistoName.str("");
+	HistoName << "ChargeOfDigisVsWidth";
+	if (Parameters.getParameter<bool>("switch") && WidthParameters.getParameter<bool>("switch"))
+	  local_mes.ChargeOfDigisVsWidth = ibooker.book2D(HistoName.str(),
+							  HistoName.str(),
+							  Parameters.getParameter<int32_t>("Nbins"),
+							  Parameters.getParameter<double>("xmin"),
+							  Parameters.getParameter<double>("xmax"),
+							  WidthParameters.getParameter<int32_t>("Nbins"),
+							  WidthParameters.getParameter<double>("xmin"),
+							  WidthParameters.getParameter<double>("xmax"));
+	else
+	  local_mes.ChargeOfDigisVsWidth = nullptr;
+      }
     }
+    
+    layerMEs.insert(std::make_pair(key, local_mes));
+  }
+}
+std::string Phase2TrackerMonitorDigi::getHistoId(uint32_t det_id, const TrackerTopology* tTopo, bool flag) {
+  int layer;
+  std::string Disc;
+  std::ostringstream fname1;
+  if(flag){
+    layer = tTopo->getITPixelLayerNumber(det_id);
+    //std::cout<<"PxLayer: "<<layer<<"\n";
+  }
+  else {
+    layer = tTopo->getOTLayerNumber(det_id);
+  }
+  if(layer < 0)
+    return "";
+
+  if (layer < 100) {
+    fname1 << "Barrel/";
+    fname1 << "Layer" << layer;
+    fname1 << "";
+  } else {
+    int side = layer / 100;
+    fname1 << "EndCap_Side" << side << "/";
+    int disc = layer - side * 100;
+    if (flag) Disc = (disc < 9) ? "FPIX_1" : "FPIX_2";
```

Comment:
Perhaps EPix (endcap pixel) and FPix (forward pixel) are more inline with official nomemclature.